src/bin/add/fetch.rs

```diff
@@ -36,20 +37,30 @@ pub fn get_latest_dependency(crate_name: &str) -> Result<Dependency, FetchVersio
     Ok(dep)
 }
 
+// Checks whether a version object is a stable release
+fn version_is_stable(version: &Object) -> bool {
+    version.get("num")
+        .and_then(Json::as_string)
+        .and_then(|s| Version::parse(s).ok())
```

Comment:
`Json::as_string` returns an `Option`, while `Version::parse` returns a `Result`, and both types don't flow nicely into each other as far as I've seen. The ok is necessary unless you want to nest closures. Or maybe you meant something else?